### PR TITLE
[intfmgr]: Support configuring IPs for all regular interfaces

### PR DESCRIPTION
**Pull request by stcheng**
Enable configuring IPv4 and IPv6 addresses on regular ports
Add tests to test IPv4 add/remove to regular ports

Signed-off-by: Shu0T1an ChenG <shuche@microsoft.com>

NOTE:
1. LAG IP configuration test will be added later once the support of LAG creation/removal feature is added.
2. IPv6 tests will be added later once the docker vs supports IPv6 after some docker configuration changes.

### CODE REVIEW

**Comment by stcheng** [comment]
@jipanyang This commit is to address the intfmgrd so that it supports the configurations via the database. I'll then remove the `/etc/network/interfaces` file in the buildimage repository and add support to configure all the IPs in the configuration database at first.

**Comment by jipanyang** [comment]
@stcheng Thanks for the explanation, change  looks good to me.

**Comment by lguohan** [comment]
do not merge until to have the buildimage pr ready to merge.

File: cfgmgr/intfmgr.cpp [hunk 1]
```diff
@@ -25,12 +25,20 @@ IntfMgr::IntfMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, c
...
     string res;
 
-    cmd << IP_CMD << " address " << opCmd << " " << ipPrefixStr << " dev " << alias;;
+    if (ipv4)
+    {
+        cmd << IP_CMD << " address " << opCmd << " " << ipPrefixStr << " dev " << alias;;
+    }
+    else
+    {
+        cmd <<IP_CMD << " -6 address " << opCmd << " " << ipPrefixStr << " dev " << alias;;
```
Comment (jleveque) [inline cfgmgr/intfmgr.cpp]: - Add space between `<<` and `IP_CMD`
- Remove extra trailing semicolon.

File: cfgmgr/intfmgr.cpp [hunk 2]
```diff
@@ -25,12 +25,20 @@ IntfMgr::IntfMgr(DBConnector *cfgDb, DBConnector *appDb, DBConnector *stateDb, c
...
+bool IntfMgr::setIntfIp(const string &alias, const string &opCmd,
+                        const string &ipPrefixStr, const bool ipv4)
 {
     stringstream cmd;
     string res;
 
-    cmd << IP_CMD << " address " << opCmd << " " << ipPrefixStr << " dev " << alias;;
+    if (ipv4)
+    {
+        cmd << IP_CMD << " address " << opCmd << " " << ipPrefixStr << " dev " << alias;;
```
Comment (jleveque) [inline cfgmgr/intfmgr.cpp]: Remove extra trailing semicolon.

File: cfgmgr/intfmgr.cpp (hunk 1 above)
Comment (stcheng) [reply cfgmgr/intfmgr.cpp]: resolved

File: cfgmgr/intfmgr.cpp (hunk 2 above)
Comment (stcheng) [reply cfgmgr/intfmgr.cpp]: resolved

**Pull request merged**